web/ui/static/js/prom_console.js

```diff
@@ -562,7 +561,7 @@ PromConsole.Graph.prototype.buildQueryUrl = function(expr) {
   var p = this.params;
   return PATH_PREFIX + "/api/v1/query_range?query=" +
     encodeURIComponent(expr) +
-    "&step=" + p.duration / this.graphTd.offsetWidth +
+    "&step=" + Math.floor(self.params.duration / this.graphTd.offsetWidth * 1000) / 1000; +
```

Comment:
```suggestion
    "&step=" + Math.floor(p.duration / this.graphTd.offsetWidth * 1000) / 1000; +
```

since we already have this

Reply:
Done